oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/OpaqueRefreshTokenOAuth2TokenIssuer.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.server.authorization.token;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Base64;
+
+import org.springframework.security.crypto.keygen.Base64StringKeyGenerator;
+import org.springframework.security.crypto.keygen.StringKeyGenerator;
+import org.springframework.security.oauth2.core.OAuth2RefreshToken;
+import org.springframework.security.oauth2.core.OAuth2RefreshToken2;
+import org.springframework.util.Assert;
+
+/**
+ * An implementation of {@link OAuth2TokenIssuer} returning an opaque {@link OAuth2RefreshToken}.
+ *
+ * @see OAuth2RefreshToken
+ * @see OAuth2TokenIssuer
+ *
+ * @author Alexey Nesterov
+ * @since 0.0.4
+ */
+public class OpaqueRefreshTokenOAuth2TokenIssuer implements OAuth2TokenIssuer<OAuth2RefreshToken> {
```

Comment:
Let's remove this. The only place we need to create an `OAuth2RefreshToken` is in `OAuth2AuthorizationCodeAuthenticationProvider` and `OAuth2RefreshTokenAuthenticationProvider`. The code is minimal so I'd rather not expose an API at this point. We could put the code in `OAuth2AuthenticationProviderUtils` to reuse if necessary.

Reply:
The only reason I think this class is useful is that it can be overriden as a bean to customize `OAuth2AuthorizationCodeAuthenticationProvider` and `OAuth2RefreshTokenAuthenticationProvider` behaviour when generating refresh token. This seems to be a reasonable extension point, even though I'm not sure users will even change that. Are you sure you want to remove it?